app/frontend/components/domains/users/invite-screen.tsx

```diff
@@ -0,0 +1,234 @@
+              <Flex direction="column" gap={6}>
+                <Flex direction="column" gap={4}>
+                  {fields.map((field, index) => (
+                    <UserInput key={field.id} index={index} remove={remove} />
+                  ))}
+                  <Button
+                    type="button"
+                    variant="tertiary"
+                    onClick={() => append(defaultUserValues)}
+                    leftIcon={<FontAwesomeIcon style={{ height: 14, width: 14 }} icon={faPlus} />}
+                  >
+                    {t("user.addUser")}
+                  </Button>
+                </Flex>
+                <Flex gap={4}>
+                  <Button
+                    variant="primary"
+                    type="submit"
+                    isLoading={isSubmitting}
+                    loadingText={t("ui.loading")}
+                    rightIcon={<FontAwesomeIcon style={{ height: 14, width: 14 }} icon={faPaperPlane} />}
+                  >
+                    {t("user.sendInvites")}
+                  </Button>
+                  <Button variant="secondary" isLoading={isSubmitting} onClick={() => navigate(-1)}>
+                    {t("ui.cancel")}
+                  </Button>
+                </Flex>
+              </Flex>
+            </form>
+          </FormProvider>
+        </Flex>
+      </Container>
+    </Flex>
+  )
+})
+
+interface INameFormControlProps {
+  label: string
+  index: number
+  subFieldName: string
+}
+
+const NameFormControl = ({ label, index, subFieldName }: INameFormControlProps) => {
+  const { register, formState } = useFormContext()
+  const { t } = useTranslation()
+
+  return (
+    <FormControl isInvalid={!!formState?.errors.users?.[index]?.[subFieldName]} flex={2}>
+      <FormLabel>{label}</FormLabel>
+      <InputGroup>
+        <Flex w="full" direction="column">
+          <Input
+            bg="greys.white"
+            {...register(`users.${index}.${subFieldName}`, {
+              validate: {
+                satisfiesNameLength: (str) => !str || (str.length >= 2 && str.length < 128) || t("ui.invalidInput"),
+              },
+            })}
+            type={"text"}
+          />
+        </Flex>
+      </InputGroup>
+    </FormControl>
+  )
+}
+
+interface IUserInputProps {
+  index: number
+  remove: (index: number) => any
+}
+
+const UserInput = observer(({ index, remove }: IUserInputProps) => {
+  const { register, formState, control, watch } = useFormContext()
+  const { t } = useTranslation()
+
+  const emailWatch = watch(`users.${index}.email`)
+
+  const { userStore } = useMst()
+  const { invitedEmails, takenEmails } = userStore
+  const invited = invitedEmails?.includes(emailWatch)
+  const taken = takenEmails?.includes(emailWatch)
+
+  return (
+    <Box bg="greys.grey03" p={2} borderRadius="sm">
+      <Flex gap={4} align="flex-end">
+        <FormControl flex={2}>
+          <FormLabel>{t("auth.role")}</FormLabel>
```

Comment:
is this label connected to the input? Might have to use for and id for accessibility